draft-ietf-wimse-client-assertion-in-workload-environments.md

```diff
@@ -0,0 +1,546 @@
++-------------------------------------------------------+
+~~~
+{: #fig-kubernetes title="OAuth2 Assertion Flow in a Kubernetes Workload Environment"}
+
+The steps shown in {{fig-kubernetes}} are:
+
+* 1) The Kubernetes Control Plane schedules the workload. This is much simplified and technically happens asynchronously.
+
+* 2) The Kubernetes Control Plane projects the service account token into the workload. This step is also much simplified and technically happens alongside the scheduling with step 1.
+
+* 3) Workloads present the projected service account token as a client assertion towards an external authorization server according to {{RFC7523}}.
+
+* 4) On success, an access token is returned to the workload to access the protected resource.
+
+* 5) The access token is used to access the protected resource in the external authorization domain.
+
+As an example, the following JSON showcases the claims a Kubernetes Service Account token carries.
+
+~~~json
+{
+  "aud": [  # matches the requested audiences, or the API server's default audiences when none are explicitly requested
+    "https://kubernetes.default.svc"
+  ],
+  "exp": 1731613413,
+  "iat": 1700077413,
+  "iss": "https://kubernetes.default.svc",  # matches the first value passed to the --service-account-issuer flag
+  "jti": "ea28ed49-2e11-4280-9ec5-bc3d1d84661a",  # ServiceAccountTokenJTI feature must be enabled for the claim to be present
+  "kubernetes.io": {
+    "namespace": "my-namespace",
+    "node": {  # ServiceAccountTokenPodNodeInfo feature must be enabled for the API server to add this node reference claim
+      "name": "127.0.0.1",
+      "uid": "58456cb0-dd00-45ed-b797-5578fdceaced"
+    },
+    "pod": {
+      "name": "my-workload-69cbfb9798-jv9gn",
+      "uid": "778a530c-b3f4-47c0-9cd5-ab018fb64f33"
+    },
+    "serviceaccount": {
+      "name": "my-workload",
+      "uid": "a087d5a0-e1dd-43ec-93ac-f13d89cd13af"
+    },
+    "warnafter": 1700081020
+  },
+  "nbf": 1700077413,
+  "sub": "system:serviceaccount:my-namespace:my-workload"
+}
+~~~
+{: #fig-kubernetes-token title="Example Kubernetes Service Account Token claims"}
+
+## Secure Production Identity Framework For Everyone (SPIFFE) {#spiffe}
+
+Secure Production Identity Framework For Everyone, also known as SPIFFE, is a cloud native compute foundation (CNCF) adopted project which defines an API defined called "Workload API" to delivery machine identity to workloads. Workloads can retrieve either X509 based or JWT credentials without the need to authenticate making it very easy to use. How workloads authenticate on the API is not part of the specification. It is common to use platform metadata from the operating system and the workload platform for authentication on the Workload API.
+
+For the scope of this document, the JWT formatted credential is the most relevant one. SPIFFE referres to it as "JWT-SVID" (JWT - Single Verifyable Identity Document).
```

Comment:
```suggestion
For the scope of this document, the JWT formatted credential is the most relevant one. SPIFFE refers to it as "JWT-SVID" (JWT - SPIFFE Verifiable Identity Document).
```
SVID is short for SPIFFE Verifiable Identity Document, see https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-verifiable-identity-document-svid

Reply:
Thank you!